src/oracle/OracleWriter.sol

```diff
@@ -17,19 +17,34 @@ abstract contract OracleWriter is Pair, IOracleWriter {
     uint256 internal constant MAX_CHANGE_PER_SEC = 0.01e18;
     string internal constant ALLOWED_CHANGE_NAME = "Shared::allowedChangePerSecond";
 
-    Observation[65536] public observations;
+    Observation[65536] internal _observations;
     uint16 public index = type(uint16).max;
 
+    mapping(address => bool) authorizedCallers;
+
     // maximum allowed rate of change of price per second
     // to mitigate oracle manipulation attacks in the face of post-merge ETH
     uint256 public allowedChangePerSecond;
     uint256 public prevClampedPrice;
 
+    modifier authorized() {
+        require(authorizedCallers[msg.sender], "OW: UNAUTHORIZED_TO_READ_ORACLE");
+        _;
+    }
+
     constructor() {
         uint256 lAllowedChangePerSecond = factory.read(ALLOWED_CHANGE_NAME).toUint256();
         setAllowedChangePerSecond(lAllowedChangePerSecond);
     }
 
+    function observations(uint256 aIndex) external view authorized returns(Observation memory rObservation) {
+        return _observations[aIndex];
+    }
+
+    function setAuthorized(address aCaller, bool aAuthorized) external onlyFactory {
+        authorizedCallers[aCaller] = aAuthorized;
+    }
```

Comment:
When you move array from a mapping model, you should then support pulling the new address from the factory, similar to fees

Reply:
Also, does this design mean we need to configure every single pair to be authorized? Seems very intense